[
    {
        "content": "<p>glacier's README says</p>\n<blockquote>\n<p>This repository is used to test internal compiler errors (also known as ICEs) in Rust. An ICE means that something went wrong, something unexpected. As such, sometimes, ICEs get randomly fixed. A compiler refactoring will remove the odd code path, even if the point of the change wasn’t to fix the bug.</p>\n<p>As such, this repository is a collection of these bugs, and it runs on Rust nightly, once a day, through GitHub Actions. If any of the ICEs stop happening, the build will fail, and I can close the associated bug.</p>\n</blockquote>\n<p>This just says it'll close the bug report, but doesn't say what happens to the regression test that was added to glacier. Will it push it to the main repo? If not, how do we know if an ICE that was unintentionally/accidentally fixed is re-introduced later?</p>",
        "id": 223087748,
        "sender_full_name": "osa1",
        "timestamp": 1610954610
    },
    {
        "content": "<p>It only checks if the bug is fixed. Nothing less, nothing more.</p>",
        "id": 223097735,
        "sender_full_name": "bjorn3",
        "timestamp": 1610962640
    },
    {
        "content": "<p>The source of glacier is really small.</p>",
        "id": 223097850,
        "sender_full_name": "bjorn3",
        "timestamp": 1610962690
    },
    {
        "content": "<p>Got it. It's a bit unfortunate that if a ICE is fixed unintentionally or accidentally we won't have a test for it.</p>",
        "id": 223105815,
        "sender_full_name": "osa1",
        "timestamp": 1610967441
    },
    {
        "content": "<p>The issue is never closed automatically</p>",
        "id": 223107664,
        "sender_full_name": "Léo Lanteri Thauvin",
        "timestamp": 1610968721
    },
    {
        "content": "<p>If we think the issue needs a test, we will add a test (manually)</p>",
        "id": 223107701,
        "sender_full_name": "Léo Lanteri Thauvin",
        "timestamp": 1610968745
    },
    {
        "content": "<p>Oh, OK. \"I can close the associated bug\" I read this as if it's closing the bug report.</p>",
        "id": 223108505,
        "sender_full_name": "osa1",
        "timestamp": 1610969272
    },
    {
        "content": "<p>So it's just adding a comment to the bug report saying that it no longer causing an ICE?</p>",
        "id": 223108571,
        "sender_full_name": "osa1",
        "timestamp": 1610969293
    },
    {
        "content": "<p>I think the bot opens a PR on glacier to move the ICE test case to a \"fixed\" directory</p>",
        "id": 223116127,
        "sender_full_name": "Léo Lanteri Thauvin",
        "timestamp": 1610974619
    },
    {
        "content": "<p>See for example glacier#617</p>",
        "id": 223116408,
        "sender_full_name": "Léo Lanteri Thauvin",
        "timestamp": 1610974792
    },
    {
        "content": "<p>Yeah, the bot doesn't close the issue; it just opens a PR to move it to fixed and then the human who merges the PR can decide to close the issue if that makes sense.</p>",
        "id": 223157351,
        "sender_full_name": "Noah Lev",
        "timestamp": 1610998647
    }
]